lib/client.js

```diff
@@ -120,6 +102,31 @@ Client.prototype.connect = function () {
   });
 };
 
+Client.prototype.setupBrokerProfiles = function (brokers) {
+  this.brokerProfiles = Object.create(null);
+  var self = this;
+
+  Object.keys(brokers).forEach(function (key, index) {
+    var brokerProfile = brokers[key];
+    var addr = brokerProfile.host + ':' + brokerProfile.port;
+    self.brokerProfiles[addr] = brokerProfile;
+    if (self.ssl) {
+      var sslAddress = brokerProfile.endpoints.find(function (endpoint) {
+        return url.parse(endpoint).protocol === 'ssl:';
+      });
+      if (!sslAddress) {
+        var err = new Error('No SSL kafka endpoint found');
```

Comment:
This method is a refactor from what @CharlesWall wrote. I have updated the error message to include the broker addr and id. I would expect all brokers in a cluster to be configured similarly to each other and it would be a big problem if one is missing an secure endpoint. Emitting an error is the right thing to do here.